File: src/test/java/ch/zizka/junitdiff/AppTest.java
package ch.zizka.junitdiff;

import junit.framework.Test;
import junit.framework.TestCase;
import junit.framework.TestSuite;

import java.io.File;

/**
 * Unit test for simple App.
 */
public class AppTest 
    extends TestCase
{
    /**
     * Create the test case
     *
     * @param testName name of the test case
     */
    public AppTest( String testName )
    {
        super( testName );
    }

    /**
     * @return the suite of tests being tested
     */
    public static Test suite()
    {
        return new TestSuite( AppTest.class );
    }

    /**
     * HTML output.
     */
    public void testHibernateRunsHTML()
    {
		System.out.println("moje: " + System.getProperty("moje")); ///
		String dataDir = System.getProperty("junitdiff.test.data.dir");

		String outPath = "target/test-tmp/"+this.getName()+"/output.html";
		File outFile = new File(outPath);
		outFile.getParentFile().mkdirs();
        
		try {
			JUnitDiffApp.main( new String[]{
				dataDir + "/hibernate-run1",
				dataDir + "/hibernate-run2",
				dataDir + "/hibernate-run3",
				dataDir + "/hibernate-run4",
				"-o", outPath
			} );
		}
		catch( Throwable t ){
			fail( t.getMessage() );
		}

		assertTrue( outFile.exists() );
    }
	
    /**
     * XML output.
     */
    public void testHibernateRunsXML()
    {
		System.out.println("moje: " + System.getProperty("moje")); ///
		String dataDir = System.getProperty("junitdiff.test.data.dir");

		String outPath = "target/test-tmp/"+this.getName()+"/output.xml";
		File outFile = new File(outPath);
		outFile.getParentFile().mkdirs();

		try {
			JUnitDiffApp.main( new String[]{
				dataDir + "/hibernate-run1",
				dataDir + "/hibernate-run2",
				dataDir + "/hibernate-run3",
				dataDir + "/hibernate-run4",
				"-o", outPath, "-xml"
			} );
		}
		catch( Throwable t ){
			fail( t.getMessage() );
		}
		
		assertTrue( outFile.exists() );
    }

}

File: src/test/java/ch/zizka/junitdiff/AppTest.kt
package ch.zizka.junitdiff

import ch.zizka.junitdiff.JUnitDiffApp.Companion.main
import junit.framework.Test
import junit.framework.TestCase
import junit.framework.TestSuite
import java.io.File

/**
 * Unit test for simple App.
 *
 * Create the test case.
 * @param testName name of the test case
 */
class AppTest (testName: String) : TestCase(testName) {

    fun testHibernateRuns_HtmlOutput() {

        val dataDir = System.getProperty("junitdiff.test.data.dir")
        val outPath = "target/test-tmp/" + this.name + "/output.html"
        val outFile = File(outPath)
        outFile.parentFile.mkdirs()

        main(
            arrayOf(
                "$dataDir/hibernate-run1",
                "$dataDir/hibernate-run2",
                "$dataDir/hibernate-run3",
                "$dataDir/hibernate-run4",
                "-o", outPath
            )
        )

        assertTrue(outFile.exists())
    }

    /**
     * XML output.
     */
    fun testHibernateRunsXML() {

        val dataDir = System.getProperty("junitdiff.test.data.dir")
        val outPath = "target/test-tmp/" + this.name + "/output.xml"
        val outFile = File(outPath)
        outFile.parentFile.mkdirs()
        try {
            main(
                arrayOf(
                    "$dataDir/hibernate-run1",
                    "$dataDir/hibernate-run2",
                    "$dataDir/hibernate-run3",
                    "$dataDir/hibernate-run4",
                    "-o", outPath, "-xml"
                )
            )
        } catch (t: Throwable) {
            kotlin.test.fail(t.message)
        }
        assertTrue(outFile.exists())
    }

    companion object {
        /**
         * @return the suite of tests being tested
         */
        fun suite(): Test {
            return TestSuite(AppTest::class.java)
        }
    }
}